### PR TITLE
Audit Fixes

### PR DESCRIPTION
**Pull request by glottologist**
This PR is to address audit issues / questions raised by @tomjack.

- Explicit validation of tokens on post and deposit.  (All fields, not just token names)
- Removal of redundant validation methods

### CODE REVIEW

**Comment by kienle371999** [comment]
LGTM.

**Comment by glottologist** [comment]
@tomjack (@kienle371999 , @Hakimba ).  I have simplified the batch logic to use the variants internally to the contract but to provide a helper to the UI .

**Comment by glottologist** [comment]
@tj825 ^^

File: batcher/batch.mligo [hunk 1]
```diff
@@ -140,49 +106,76 @@ let close (batch : t) : t =
...
-    | Closed _ ->
-      finalize batch current_time clearing rate
-    | _ -> failwith Errors.trying_to_finalize_batch_which_is_not_closed
 
 let new_batch_set : batch_set =
   {
-    current_batch_number = 0n;
-    last_batch_number= 0n;
+    current_batch_index = 0n;
+    current_batch_status = 2n;
```
[the remark this answers is not in the document]
Comment (glottologist) [reply batcher/batch.mligo:113]: This is a helper for the UI to easily check the batch status.  We will still need to deal with the variant to get the remaining time left in the batch but it should make it easier

[the remark this answers is not in the document]
Comment (glottologist) [reply batcher/batch.mligo:113]: We can remove it I think.  I am looking at the UI code now and we query the batch directly for the status and now the changes to the batch mechanism make this redundant as we can always inspect a current batch.  I'll remove

[the remark this answers is not in the document]
Comment (glottologist) [reply batcher/batch.mligo]: Removed

**Pull request merged**